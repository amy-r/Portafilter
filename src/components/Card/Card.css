@import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400');


.roaster-name {
  color: white;
  text-transform: uppercase;
  font-family: 'Open Sans', sans-serif;
  letter-spacing: 0.2rem;
  margin-bottom: 20px;
}

.roaster-card {
  height: 320px;
  width: 200px;
  background-color: #aa9e91;
  border-radius: 10px;
  padding: 20px;
  position: relative;
  margin: 20px 20px;
  box-shadow: 0px 0px 3px 0px rgba(119, 119, 119, 0.63);
  -moz-box-shadow: 0px 0px 3px 0px rgba(119, 119, 119, 0.63);
  -webkit-box-shadow: 0px 0px 3px 0px rgba(119, 119, 119, 0.63);
}

#roaster-list {
  margin: auto;
  list-style: none;
  padding: 0;
}

.roaster-data {
  font-family: 'Open Sans', sans-serif;
  color: #444444;
}

.new-coffee {
  background-color: #444444;
  color: white;
  padding: 10px;
  position: absolute;
  bottom: 10px;
  left: 125px;
  height: 50px;
  width: 100px;
  display: block;
  border-style: none;
  -webkit-box-shadow: 0px 0px 28px -3px rgba(0,0,0,0.75);
-moz-box-shadow: 60px 0px 28px -3px rgba(0,0,0,0.75);
box-shadow: 0px 0px 28px -3px rgba(0,0,0,0.75);
}

.form-button:hover, .new-coffee:hover {
  cursor: pointer;
}

.form-button {
  background-color: #444444;
  color: white;
  padding: 10px;
  position: absolute;
  bottom: 10px;
  left: 15px;
  height: 50px;
  width: 100px;
  display: block;
  border-style: none;
  -webkit-box-shadow: 0px 0px 28px -3px rgba(0,0,0,0.75);
-moz-box-shadow: 60px 0px 28px -3px rgba(0,0,0,0.75);
box-shadow: 0px 0px 28px -3px rgba(0,0,0,0.75);
}