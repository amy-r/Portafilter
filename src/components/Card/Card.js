import React from 'react';
import ReactDOM from 'react-dom';

export const Card = (props) => {
  const card = Object.entries(props).map ( entry => {
    return <li> { entry[0] } : { entry[1] } </li>
  })

  return(
    <div>
      {card}
    </div>
  )
}
