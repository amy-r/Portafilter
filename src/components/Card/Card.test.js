import React from 'react';
import ReactDOM from 'react-dom';
import { Card } from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Card />)

    expect(wrapper).toMatchSnapshot();
  })
})

