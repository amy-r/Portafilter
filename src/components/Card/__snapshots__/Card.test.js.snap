// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="roaster-card"
>
        <ul
                id="roaster-list"
        >
                
        </ul>
        <Link
                replace={false}
                to="coffee-form"
        >
                 
                <button
                        className="new-coffee"
                >
                        + NEW COFFEE
                </button>
                 
        </Link>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ul
            id="roaster-list"
>
            
</ul>,
          <Link
            replace={false}
            to="coffee-form"
>
             
            <button
                        className="new-coffee"
            >
                        + NEW COFFEE
            </button>
             
</Link>,
        ],
        "className": "roaster-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "id": "roaster-list",
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              " ",
              <button
                className="new-coffee"
>
                + NEW COFFEE
</button>,
              " ",
            ],
            "replace": false,
            "to": "coffee-form",
          },
          "ref": null,
          "rendered": Array [
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "+ NEW COFFEE",
                "className": "new-coffee",
              },
              "ref": null,
              "rendered": "+ NEW COFFEE",
              "type": "button",
            },
            " ",
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="roaster-card"
>
          <ul
                    id="roaster-list"
          >
                    
          </ul>
          <Link
                    replace={false}
                    to="coffee-form"
          >
                     
                    <button
                              className="new-coffee"
                    >
                              + NEW COFFEE
                    </button>
                     
          </Link>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ul
              id="roaster-list"
>
              
</ul>,
            <Link
              replace={false}
              to="coffee-form"
>
               
              <button
                            className="new-coffee"
              >
                            + NEW COFFEE
              </button>
               
</Link>,
          ],
          "className": "roaster-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "id": "roaster-list",
            },
            "ref": null,
            "rendered": Array [],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                " ",
                <button
                  className="new-coffee"
>
                  + NEW COFFEE
</button>,
                " ",
              ],
              "replace": false,
              "to": "coffee-form",
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "+ NEW COFFEE",
                  "className": "new-coffee",
                },
                "ref": null,
                "rendered": "+ NEW COFFEE",
                "type": "button",
              },
              " ",
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
